llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -659,17 +639,20 @@ Value *VPInstruction::generate(VPTransformState &State) {
   }
   case VPInstruction::BranchOnCond: {
     Value *Cond = State.get(getOperand(0), VPLane(0));
-    auto *Br = createCondBranch(Cond, getParent(), State);
+    // Replace the temporary unreachable terminator with a new conditional
+    // branch, hooking it up to backward destination for latch blocks now, and
+    // to forward destination(s) later when they are created.
+    VPBasicBlock *SecondVPSucc =
```

Comment:
Does the "Second successor may be backwards - iff it is already in VPBB2IRBB." comment still hold?